spotbugs/src/main/java/edu/umd/cs/findbugs/ba/ResourceValueFrame.java

```diff
@@ -37,20 +37,31 @@ public class ResourceValueFrame extends Frame<ResourceValue> {
      */
     public static final int OPEN_ON_EXCEPTION_PATH = 2;
 
+    /**
+     * The resource is closed without being opened in the first place.
+     */
+    public static final int CLOSED_WITHOUT_OPENED = 3;
+
     /**
      * The resource is closed (or unlocked, etc).
      */
-    public static final int CLOSED = 3;
```

Comment:
better to keep public APIs compatible. Can we move new constants after the `NONEXISTENT`?

Reply:
Yes, will do.

Reply:
Unfortunately no, because the mergeing of these states are based on a `Math.min()` call. This would mean, that a `CLOSED` would override a `CLOSED_WITHOUT_OPENED` state. Then the information that the lock was not even acquired in the first place would be lost.

Reply:
Hi @KengoTODA could you revisit this PR and check my comment above?

Reply:
Then it is a breaking change, and I think only should be merged for 5.0.0, together with #2069, which may cause some merge conflicts with this one. Also, it would be worth to mention the breaking change in the changelog as well.